_faq/cep-and-its-importance.md

```diff
@@ -0,0 +1,12 @@
+---
+question: "What is a cEP and why it is so important to do analysis pre-gsoc and design as early as possible, to be finalized in the cEP?"
+---
+A **cEP (coala Enhancement Proposal)** is proposed to the coala community to discuss the new major features that the contributor intends to implement during GSoC.
+
+**cEP's** are designed before implementation. Implementing major features usually happens from a technical perspective and may lead to poorly designed content - developer and user wise. This can essentially save a lot of coding work and also makes sure that major features are properly documented.
+
+Experience tells us that the best GSoC students are the ones who heavily contributed to the project ideas website, created projects from the issues, and expanded on existing ***.md*** files.
+
+So, if you are planning to do GSoC with coala, you should be starting to write detailed project ideas in our [projects](https://github.com/coala/projects) repository. Also, you should learn how to express complex technical ideas in the markup because if you can't explain your idea, and how you will achieve it, we probably won't be able to mentor you.
```

Comment:
Not sure about this but can we have something else (or nothing at all!) in place of `we probably won't be able to mentor you.`

Reply:
I will think upon the last line and will look to reword that line that looks good.
Will commit as soon as I get time, busy with exams study 😞

Reply:
@utkarsh2102 `we probably won't be able to select your proposal` in place of `we probably won't be able to mentor you` , will it be ok ? or something else ?

Reply:
Can we not have that line at all? :P
Not sure. @jayvdb would be the right person to tell you.
CC: @li-boxuan.

Reply:
IMO that line should be there but it can be expressed in more appropriate way because the contributor must know that the cEP he is working on has to be in detailed manner with proper markup tools used at right places and should be represented good, so that mentors can easily understand what will that contributor intends to implement.
Will wait for maintainers to get their view 👍

Reply:
> IMO that line should be there

I disagree. Everyone applying for gsoc know the possible outcome of either getting selected or rejected. The way @KVGarg has mentioned it would make us seem like a community that is too hard to get into which is far from what this community is actually like. I have not seen a community where the maintainers and the admin try so hard to select every good student for GSoC and even advise them to submit multiple proposals to other organizations.

So I am with @utkarsh2102 on this. It should be rephrased like *an interested student must do enough research about their project to be able to express their ideas, timeline, etc clearly in their GSoC proposal which will increase their chances of success*.

What you ( @KVGarg  ) wrote is absolutely correct but I just don't find it very welcoming towards newcomers.

Reply:
Thnx @palash25 for your review 
And i have totally understand what you are trying to say and will rephrase that last line as like you have said

Reply:
hey @palash25 can you please review the latest commit